spyder/plugins/application/container.py

```diff
@@ -99,6 +105,10 @@ def __init__(self, name, plugin, parent=None):
     # ---- PluginMainContainer API
     # -------------------------------------------------------------------------
     def setup(self):
+
+        self.application_update_status = ApplicationUpdateStatus(parent=self)
```

Comment:
To use the check_updates method from the status bar taking into account the other changes then here we just need to do the signal connection:
```suggestion
        self.application_update_status = ApplicationUpdateStatus(parent=self)
        self.application_update_status.sig_check_for_updates_requested.connect(
            self.check_updates
        )
```